ramp-engine/ramp_engine/tests/test_conda_worker.py

```diff
@@ -103,80 +115,96 @@ def test_conda_worker(submission, get_conda_worker):
                                            'training_output')
         assert not os.path.exists(output_training_dir), \
             "teardown() failed to remove the predictions"
-    finally:
-        # remove all directories that we potentially created
-        _remove_directory(worker)
-
-
-def test_conda_worker_without_conda_env_specified(get_conda_worker):
-    worker = get_conda_worker('starting_kit')
-    # remove the conda_env parameter from the configuration
-    del worker.config['conda_env']
-    # if the conda environment is not given in the configuration, we should
-    # fall back on the base environment of conda
-    # the conda environment is set during setup; thus no need to launch
-    # submission
-    worker.setup()
-    assert 'envs' not in worker._python_bin_path
-
-
-def test_conda_worker_error_missing_config_param(get_conda_worker):
-    worker = get_conda_worker('starting_kit')
-    # we remove one of the required parameter
-    del worker.config['kit_dir']
-
-    err_msg = "The worker required the parameter 'kit_dir'"
-    with pytest.raises(ValueError, match=err_msg):
-        worker.setup()
 
 
-def test_conda_worker_error_unknown_env(get_conda_worker):
-    worker = get_conda_worker('starting_kit', conda_env='xxx')
-    msg_err = "The specified conda environment xxx does not exist."
-    with pytest.raises(ValueError, match=msg_err):
+@pytest.mark.parametrize("Worker", ALL_WORKERS)
+def test_conda_worker_without_conda_env_specified(Worker, dask_scheduler):
+    with get_conda_worker(
+        'starting_kit', Worker=Worker, dask_scheduler=dask_scheduler
+    ) as worker:
+        # remove the conda_env parameter from the configuration
+        del worker.config['conda_env']
+        # if the conda environment is not given in the configuration, we should
+        # fall back on the base environment of conda
+        # the conda environment is set during setup; thus no need to launch
+        # submission
         worker.setup()
-        assert worker.status == 'error'
+        assert 'envs' not in worker._python_bin_path
+
+
+@pytest.mark.parametrize("Worker", ALL_WORKERS)
+def test_conda_worker_error_missing_config_param(Worker, dask_scheduler):
+    with get_conda_worker(
+        'starting_kit', Worker=Worker, dask_scheduler=dask_scheduler
+    ) as worker:
+        # we remove one of the required parameter
+        del worker.config['kit_dir']
+
+        err_msg = "The worker required the parameter 'kit_dir'"
+        with pytest.raises(ValueError, match=err_msg):
+            worker.setup()
+
 
+@pytest.mark.parametrize("Worker", ALL_WORKERS)
+def test_conda_worker_error_unknown_env(Worker, dask_scheduler):
+    with get_conda_worker(
```

Comment:
maybe add a conda_env = 'xxx' variable used in the context manager declaration and in the msg_err